src/Orleans/LogViews/LogViewGrain.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System.Threading.Tasks;
+using Orleans.Concurrency;
+using Orleans.MultiCluster;
+using System;
+using System.Collections.Generic;
+using Orleans.LogViews;
+
+namespace Orleans
+{
+
+    /// <summary>
+    /// Log view grain.
+    /// <typeparam name="TView">The type for the log view, i.e. state of this grain.</typeparam>
+    /// <typeparam name="TLogEntry">The type for log entries.</typeparam>
+    /// </summary>
+    public abstract class LogViewGrain<TView, TLogEntry> :
```

Comment:
There are additional alternatives to the LogViewGrain class (QueuedGrain, JournaledGrain, not in this PR) that inherit from LogViewGrainBase but hide the log-view API underneath a terminology that is more specific to the intended situation. 

Such classes can be defined by the user, too - it gives them a chance to control how exactly the storage API should look like. If they inherit from LogViewGrain, they cannot hide the logview api.